controllers/databasecluster_controller.go

```diff
@@ -2009,8 +2097,8 @@ func (r *DatabaseClusterReconciler) reconcileLabels(ctx context.Context, databas
 		needsUpdate = true
 	}
 	if database.Spec.DataSource != nil {
-		if _, ok := database.ObjectMeta.Labels[fmt.Sprintf(backupStorageNameLabelTmpl, database.Spec.DataSource.BackupStorageName)]; !ok {
-			database.ObjectMeta.Labels[fmt.Sprintf(backupStorageNameLabelTmpl, database.Spec.DataSource.BackupStorageName)] = backupStorageLabelValue
+		if _, ok := database.ObjectMeta.Labels[fmt.Sprintf(backupStorageNameLabelTmpl, database.Spec.DataSource.DBClusterBackupName)]; !ok {
+			database.ObjectMeta.Labels[fmt.Sprintf(backupStorageNameLabelTmpl, database.Spec.DataSource.DBClusterBackupName)] = backupStorageLabelValue
```

Comment:
Posting a comment hoping to figure out if the change served to a purpose or it was unintended.

Reply:
ohhh boy, over 1 year old PR :sweat_smile:
this was done in a commit called "[Fix labels](https://github.com/percona/everest-operator/pull/51/commits/363113d3c9c29af898ea73dba5f2726e86081af8)" so it was definitely intentional but I can't recall why and now it just feels wrong :see_no_evil:

Reply:
Iirc it was done for faster filtering backups/restores per db cluster because of k8s API capabilities to filter by labels

Reply:
So. After digging it a bit deeper I remembered the following stuff:
1. It was done in the first place for faster filtering of database clusters to delete/update backups on the everest backend
2. During the previous review we forgot to use `Spec.DataSource.BackupStorageName` instead of `Spec.DataSource.DBClusterBackupName` which is a mistake here
3. On the everest-backend there was a check and the backend was fetching db clusters from k8s using labels for [faster search](https://github.com/percona/percona-everest-backend/blob/main/pkg/kubernetes/backup_storage.go#L58)
4. However, this use of labels was gone during the move to a monorepo or to something else. You can take a look here https://github.com/percona/everest/blob/main/pkg/kubernetes/backup_storage.go#L63. Using labels was intended for faster/better way of finding db clusters that use a backup storage without looping through all database clusters
I hope this sheds some light on  the situation

Reply:
Thanks @gen1us2k! Created an overhead ticket to review & improve the labelling policy EVEREST-1654